_licenses/cecill-2.1.txt

```diff
@@ -0,0 +1,281 @@
+---
+title: CeCILL Free Software License Agreement v2.1
+spdx-id: CECILL-2.1
+source: http://www.cecill.info/licences/Licence_CeCILL_V2.1-fr.html
+
+description: French open-source license made by the CEA, INRIA and CNRS, compatible with the GNU GPL v2.0 and v3.0
+
+how: Create a text file (typically named LICENSE or LICENSE.txt) in the root of your source code and copy the text of the license into the file. 
+
+using:
```

Comment:
These aren't ideal examples. The first is a license file, but due to multiple licenses and license files, wouldn't be automatically recognized. The second and third just state that the projects are licensed on the project websites. See https://github.com/github/choosealicense.com/blob/gh-pages/CONTRIBUTING.md#adding-a-license

> Identification of 3 notable projects using the license with straightforward LICENSE files which serve as examples newcomers can follow and that could be detected by licensee if it knew about the license

Are there more straightforward examples with single LICENSE files that new adopters could copy and have their license detected?

Reply:
This (3 notable projects which could be detected by licensee) is the sole remaining thing needed to merge.

Reply:
Here are three projects from the french administration using one of the CeCILL licenses:

- https://github.com/etalab/taxe-habitation
- https://github.com/etalab/calculette-impots-m-source-code
- https://github.com/InseeFr/Mesange

These projects are published by the french ministry of economy and finance - they may not not be "notable" in terms of Github fame, but they are important in terms of opengov/transparency.

I let @ljoubert update his PR, I hope this will be enough for it to be merged.

Reply:
Sorry, replying to myself: the examples already provided by @ljoubert are throwing errors in the CI.

@mlinksva can you guide us in what could be wrong? Shall we modify the URLs or the text of the licenses to pass the tests in the CI?

Reply:
@bzg those examples look reasonable to me. The first one is even fairly popular it seems. Of the existing examples I think only the BMC-tools one is workable. It was passing. It looks like the reason it might not be now is encoding, but I'm not sure what changed.

Reply:
Okay, thanks again. I'll discuss this with @ljoubert and follow up later on.

Reply:
I've spent some time digging into this issue again.

My understanding is that the CI tests fail because the [Licensee gem](https://github.com/licensee/licensee) does not detect the CeCILL v2.1 - which seems obvious because Licensee relies on... choosealicense.com licenses.

In general, I think CeCILL licenses may be hard to detect, so I'd suggest to add an exception in [https://github.com/github/choosealicense.com/blob/gh-pages/spec/license_meta_spec.rb](license_meta_spec.rb), along with the ncsa and postgresql exceptions.

WDYT?

Reply:
The tests here [monkey patch licensee](https://github.com/github/choosealicense.com/blob/791e3fe05accab0e941a9086c8086c7f42152cc7/spec/spec_helper.rb#L135-L144) to use the license texts available in the branch being tested rather than the ones that licensee vendors. 

I'm not sure why CeCILL licenses should be hard to detect; they aren't templated. Is there something about the text that I'm missing?

Reply:
I ruled out the encoding issue, I get the same [CI test failure](https://travis-ci.org/github/choosealicense.com/builds/520862523?utm_source=github_status&utm_medium=notification) with a correct encoding.  I'm out of ideas on how to interpret the failure right now.

Reply:
Thanks for researching @bzg. I'll try to dig in more when I have some free time.

Reply:
Hi @mlinksva, did you have time to find out what causes the CI test failure?

Reply:
I didn't @bzg thanks for the ping. The bmc-tools and Test-Compta-Demat examples are https://en.wikipedia.org/wiki/ISO/IEC_8859 encoded. I suspect that may be a problem for licensee. But I'm not certain as the vitam example is UTF-8 and an exact match for the text in this PR, but still failing in the test. Which, re-reading your comments above, is exactly the conclusion you came to. 😆In other words, still some debugging to do in the guts of licensee, apologies for this non-update. I'll try to dig some more tonight, feel free to bug me again if I don't say more soon.

Reply:
Alright, I confirmed that ISO-8859 encoding does cause problems, so bcm-tools and Test-Compta-Demat examples need to change (different repos, or change encoding of those files).

However, the file in vitam does get detected if I add CECILL and (and it as a test fixture) to licensee locally. So there's still something else going on in this repository's tests.

Reply:
Alright again, the test here wasn't retrieving the vitam license file correctly since the example is on a branch with numbers in its name. Fixed the test in 4b67edf. Now the vitam example will pass and there will still be 2 test failures for the other 2 examples.

Reply:
Thanks for the careful and useful tests! I've opened two PRs, fixing encoding for the license files for [Test-Compta-Demat](https://github.com/DGFiP/Test-Compta-Demat/pull/22) and [bmc-tools](https://github.com/ANSSI-FR/bmc-tools/pull/6). If this does not get merge soon enough, we can update the examples URLs and use the github.com/DISIC/ repositories. Thanks again!

Reply:
I've waited a bit but I don't think it is reasonable to wait for a merge of [this PR](https://github.com/DGFiP/Test-Compta-Demat/pull/22). I suggest that @ljoubert updates his PR using https://github.com/DISIC/Test-Compta-Demat instead, which does not have the encoding problem.

Reply:
https://github.com/DISIC/Test-Compta-Demat is just a fork of https://github.com/DGFiP/Test-Compta-Demat rather than another bona fide example. There's got to be another use on github.com or elsewhere publicly accessible.